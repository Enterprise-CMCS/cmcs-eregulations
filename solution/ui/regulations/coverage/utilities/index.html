
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for utilities</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../index.html">All files</a> utilities</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">36.78% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>487/1324</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">52.38% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>11/21</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">5.71% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>4/70</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">36.78% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>487/1324</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line low'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="api.js"><a href="api.js.html">api.js</a></td>
	<td data-value="45.47" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 45%"></div><div class="cover-empty" style="width: 55%"></div></div>
	</td>
	<td data-value="45.47" class="pct low">45.47%</td>
	<td data-value="541" class="abs low">246/541</td>
	<td data-value="52.38" class="pct medium">52.38%</td>
	<td data-value="21" class="abs medium">11/21</td>
	<td data-value="12.9" class="pct low">12.9%</td>
	<td data-value="31" class="abs low">4/31</td>
	<td data-value="45.47" class="pct low">45.47%</td>
	<td data-value="541" class="abs low">246/541</td>
	</tr>

<tr>
	<td class="file low" data-value="utils.js"><a href="utils.js.html">utils.js</a></td>
	<td data-value="30.77" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 30%"></div><div class="cover-empty" style="width: 70%"></div></div>
	</td>
	<td data-value="30.77" class="pct low">30.77%</td>
	<td data-value="783" class="abs low">241/783</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="39" class="abs low">0/39</td>
	<td data-value="30.77" class="pct low">30.77%</td>
	<td data-value="783" class="abs low">241/783</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2023-05-23T17:56:42.190Z
            </div>
        <script src="../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../sorter.js"></script>
        <script src="../block-navigation.js"></script>
    </body>
</html>
    