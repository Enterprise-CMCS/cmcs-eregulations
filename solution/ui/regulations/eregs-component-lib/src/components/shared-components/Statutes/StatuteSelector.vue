<script setup>
import { titleSelectorList } from "./schemas/listSchemas";

const props = defineProps({
    loading: {
        type: Boolean,
        required: false,
        default: false,
    },
    selectedAct: {
        type: String,
        required: false,
        default: "ssa",
    },
    selectedTitle: {
        type: String,
        required: false,
        default: "19",
    },
});

const isActActive = ({ act }) => act === props.selectedAct;
const isTitleActive = ({ act, title }) =>
    act === props.selectedAct && title === props.selectedTitle;
</script>

<template>
    <ul class="acts__list">
        <li
            v-for="(value, key, i) in titleSelectorList"
            :key="`${key}-${i}`"
            class="acts-list__item"
        >
            <h4
                class="acts-item__heading"
                :class="{
                    'acts-item__heading--active': isActActive({ act: key }),
                }"
            >
                {{ value.name }}
            </h4>
            <ul class="titles__list">
                <li
                    v-for="(title, j) in value.titles"
                    :key="`${title.title}-${j}`"
                    class="titles-list__item"
                >
                    <h4>
                        <RouterLink
                            class="titles-list__link"
                            :data-testid="`${key}-${title.titleRoman}-${title.title}`"
                            :class="{
                                'titles-list__link--active': isTitleActive({
                                    act: key,
                                    title: title.title,
                                }),
                                'titles-list__link--loading': loading,
                            }"
                            :to="{
                                name: 'statutes',
                                query: {
                                    act: key,
                                    title: title.title,
                                },
                            }"
                        >
                            Title {{ title.titleRoman }}
                        </RouterLink>
                    </h4>
                </li>
            </ul>
        </li>
    </ul>
</template>
