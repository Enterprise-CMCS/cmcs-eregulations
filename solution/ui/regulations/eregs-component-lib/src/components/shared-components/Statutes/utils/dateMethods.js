 
 

import _isNull from "lodash/isNull";
import _isUndefined from "lodash/isUndefined";

// SSA table column date methods

/**
 * @param {string} type - Label for the date
 * @param {string} date - Date to be formatted
 *
 * @returns {string} - Formatted date label or empty string
 */
const getDateLabel = ({ type, date }) => {
    if (
        _isNull(type) ||
        _isNull(date) ||
        _isUndefined(type) ||
        _isUndefined(date) ||
        type === ""
    )
        return "";

    const rawDate = new Date(date);
    // add a day to get correct month because Javascript dates are weird
    rawDate.setDate(rawDate.getDate() + 1);
    const formattedDate = rawDate.toLocaleString("default", {
        month: "short",
        year: "numeric",
    });

    if (formattedDate === "Invalid Date") return "";

    return `${type} ${formattedDate}`;
};

export { getDateLabel };
