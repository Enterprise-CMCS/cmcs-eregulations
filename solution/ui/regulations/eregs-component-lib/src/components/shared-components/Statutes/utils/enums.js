const DISPLAY_TYPES = ["table", "list"];
const TABLE_TYPES = ["aca", "ssa"];

export {
    DISPLAY_TYPES,
    TABLE_TYPES,
};
