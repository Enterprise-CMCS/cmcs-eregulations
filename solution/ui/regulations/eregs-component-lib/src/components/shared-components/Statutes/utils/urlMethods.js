 
 

// SSA table URL methods

/**
 * @param {number} title - CFR title (ex: 42)
 * @param {string} usc - USC section
 *
 * @returns {string} url - url to the House.gov page for the USC section
 */
const houseGovUrl = ({ title, usc }) =>
    `https://uscode.house.gov/view.xhtml?hl=false&edition=prelim&req=granuleid%3AUSC-prelim-title${title}-section${usc}`;

/**
 * @param {string} souce_url - URL containing COMPS number
 *
 * @returns {string | null} url - url to GovInfo.gov page displaying Statute Compilation PDF
 */
const statuteCompilationUrl = ({ source_url }) => {
    if (!source_url) return null;

    const compsNumber = source_url
        .split("/")
        .find((str) => str.includes("COMPS"));

    return `https://www.govinfo.gov/content/pkg/${compsNumber}/pdf/${compsNumber}.pdf`;
};

/**
 * @param {number} statute_title - integer representation of roman numeral Statute Title
 * @param {string} section - SSA section
 *
 * @returns {string} url - url to the SSA.gov page for the SSA section
 */
const ssaGovUrl = ({ statute_title, section }) => {
    const title = statute_title == "16" ? "16b" : statute_title;

    return `https://www.ssa.gov/OP_Home/ssact/title${title}/${section}.htm`;
};

/**
 * @param {number} title - CFR title (ex: 42)
 * @param {string} usc - USC section
 *
 * @returns {string} url - url to GovInfo.gov PDF page for USC Code
 */
const usCodeUrl = ({ title, usc }) =>
    `https://www.govinfo.gov/link/uscode/${title}/${usc}`;

export { houseGovUrl, usCodeUrl, statuteCompilationUrl, ssaGovUrl };
