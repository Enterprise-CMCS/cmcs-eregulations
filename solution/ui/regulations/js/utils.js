/* eslint-disable */
import _delay from "lodash/delay";
import _get from "lodash/get";

// a promise friendly delay function
function delay(seconds) {
    return new Promise((resolve) => {
        _delay(resolve, seconds * 1000);
    });
}

function parseError(err) {
    console.log(err);
    const errMessage = err.errors
        ? err.errors[Object.keys(err.errors)[0]][0]
        : err.message;
    errMessage && alert(errMessage);

    const message = errMessage;
    try {
        const code = Object.keys(err.errors)[0];
        const status = _get(err, "status");
        const requestId = _get(err, "requestId");
        const error = new Error(message);
        error.code = code;
        error.requestId = requestId;
        error.root = err;
        error.status = status;

        return error;
    } catch {
        return new Error(message);
    }
}

const EventCodes = {
    SetSection: 'SetSection'
}

export {
    delay,
    parseError,
    EventCodes
};
