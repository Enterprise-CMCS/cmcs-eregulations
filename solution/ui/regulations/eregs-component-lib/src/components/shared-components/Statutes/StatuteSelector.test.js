import flushPromises from "flush-promises";
import { render, screen } from "@testing-library/vue";
import { describe, it, expect } from "vitest";

import actsFixture from "cypress/fixtures/acts.json";

import { ACT_TYPES } from "sharedComponents/Statutes/utils/enums.js";
import { shapeTitlesResponse } from "utilities/utils";

import StatuteSelector from "./StatuteSelector.vue";

const SHAPED_TITLES = shapeTitlesResponse({
    actsResults: actsFixture,
    actTypes: ACT_TYPES,
});

describe("Statute Table Selector", () => {
    describe("SSA table type", () => {
        it(`Creates a snapshot of the Statute Selector with default props`, async () => {
            const wrapper = render(StatuteSelector, {
                props: {
                    titles: SHAPED_TITLES,
                },
                stubs: { RouterLink: true },
            });

            await flushPromises();

            const activeLink = screen.getByTestId("ssa-XIX-19");
            expect(
                activeLink.classList.contains("titles-list__link--active")
            ).toBe(true);
            expect(
                activeLink.classList.contains("titles-list__link--loading")
            ).toBe(false);

            const inactiveLink = screen.getByTestId("ssa-XXI-21");
            expect(
                inactiveLink.classList.contains("titles-list__link--active")
            ).toBe(false);
            expect(
                inactiveLink.classList.contains("titles-list__link--loading")
            ).toBe(false);

            expect(wrapper).toMatchSnapshot();
        });

        it(`Creates a snapshot of the Statute Selector with a loading prop`, async () => {
            const wrapper = render(StatuteSelector, {
                stubs: { RouterLink: true },
                props: {
                    loading: true,
                    titles: SHAPED_TITLES,
                },
            });

            await flushPromises();

            const activeLink = screen.getByTestId("ssa-XIX-19");
            expect(
                activeLink.classList.contains("titles-list__link--loading")
            ).toBe(true);

            const inactiveLink = screen.getByTestId("ssa-XXI-21");
            expect(
                inactiveLink.classList.contains("titles-list__link--active")
            ).toBe(false);

            expect(wrapper).toMatchSnapshot();
        });

        it(`Creates a snapshot of the Statute Selector when act and title props passed in to component`, async () => {
            const wrapper = render(StatuteSelector, {
                stubs: { RouterLink: true },
                props: {
                    selectedAct: "ssa",
                    selectedTitle: "21",
                    titles: SHAPED_TITLES,
                },
            });

            await flushPromises();

            const activeLink = screen.getByTestId("ssa-XXI-21");
            expect(
                activeLink.classList.contains("titles-list__link--active")
            ).toBe(true);

            const inactiveLink = screen.getByTestId("ssa-XIX-19");
            expect(
                inactiveLink.classList.contains("titles-list__link--active")
            ).toBe(false);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
