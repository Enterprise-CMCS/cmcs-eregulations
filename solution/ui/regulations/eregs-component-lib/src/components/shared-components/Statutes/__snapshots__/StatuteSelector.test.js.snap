// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Statute Table Selector > SSA table type > Creates a snapshot of the Statute Selector when act and title props passed in to component 1`] = `
{
  "baseElement": <body>
    <div>
      <ul
        class="acts__list"
      >
        <li
          class="acts-list__item"
        >
          <h4
            class="acts-item__heading acts-item__heading--active"
          >
             Social Security Act 
          </h4>
          <ul
            class="titles__list"
          >
            <li
              class="titles-list__item"
            >
              <h4>
                <routerlink-stub
                  class="titles-list__link"
                  data-testid="ssa-XI-11"
                  to="[object Object]"
                >
                   Title XI 
                </routerlink-stub>
              </h4>
            </li>
            <li
              class="titles-list__item"
            >
              <h4>
                <routerlink-stub
                  class="titles-list__link"
                  data-testid="ssa-XVI-16"
                  to="[object Object]"
                >
                   Title XVI 
                </routerlink-stub>
              </h4>
            </li>
            <li
              class="titles-list__item"
            >
              <h4>
                <routerlink-stub
                  class="titles-list__link"
                  data-testid="ssa-XVIII-18"
                  to="[object Object]"
                >
                   Title XVIII 
                </routerlink-stub>
              </h4>
            </li>
            <li
              class="titles-list__item"
            >
              <h4>
                <routerlink-stub
                  class="titles-list__link"
                  data-testid="ssa-XIX-19"
                  to="[object Object]"
                >
                   Title XIX 
                </routerlink-stub>
              </h4>
            </li>
            <li
              class="titles-list__item"
            >
              <h4>
                <routerlink-stub
                  class="titles-list__link titles-list__link--active"
                  data-testid="ssa-XXI-21"
                  to="[object Object]"
                >
                   Title XXI 
                </routerlink-stub>
              </h4>
            </li>
          </ul>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="acts__list"
    >
      <li
        class="acts-list__item"
      >
        <h4
          class="acts-item__heading acts-item__heading--active"
        >
           Social Security Act 
        </h4>
        <ul
          class="titles__list"
        >
          <li
            class="titles-list__item"
          >
            <h4>
              <routerlink-stub
                class="titles-list__link"
                data-testid="ssa-XI-11"
                to="[object Object]"
              >
                 Title XI 
              </routerlink-stub>
            </h4>
          </li>
          <li
            class="titles-list__item"
          >
            <h4>
              <routerlink-stub
                class="titles-list__link"
                data-testid="ssa-XVI-16"
                to="[object Object]"
              >
                 Title XVI 
              </routerlink-stub>
            </h4>
          </li>
          <li
            class="titles-list__item"
          >
            <h4>
              <routerlink-stub
                class="titles-list__link"
                data-testid="ssa-XVIII-18"
                to="[object Object]"
              >
                 Title XVIII 
              </routerlink-stub>
            </h4>
          </li>
          <li
            class="titles-list__item"
          >
            <h4>
              <routerlink-stub
                class="titles-list__link"
                data-testid="ssa-XIX-19"
                to="[object Object]"
              >
                 Title XIX 
              </routerlink-stub>
            </h4>
          </li>
          <li
            class="titles-list__item"
          >
            <h4>
              <routerlink-stub
                class="titles-list__link titles-list__link--active"
                data-testid="ssa-XXI-21"
                to="[object Object]"
              >
                 Title XXI 
              </routerlink-stub>
            </h4>
          </li>
        </ul>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "isUnmounted": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "unmount": [Function],
  "updateProps": [Function],
}
`;

exports[`Statute Table Selector > SSA table type > Creates a snapshot of the Statute Selector with a loading prop 1`] = `
{
  "baseElement": <body>
    <div>
      <ul
        class="acts__list"
      >
        <li
          class="acts-list__item"
        >
          <h4
            class="acts-item__heading acts-item__heading--active"
          >
             Social Security Act 
          </h4>
          <ul
            class="titles__list"
          >
            <li
              class="titles-list__item"
            >
              <h4>
                <routerlink-stub
                  class="titles-list__link titles-list__link--loading"
                  data-testid="ssa-XI-11"
                  to="[object Object]"
                >
                   Title XI 
                </routerlink-stub>
              </h4>
            </li>
            <li
              class="titles-list__item"
            >
              <h4>
                <routerlink-stub
                  class="titles-list__link titles-list__link--loading"
                  data-testid="ssa-XVI-16"
                  to="[object Object]"
                >
                   Title XVI 
                </routerlink-stub>
              </h4>
            </li>
            <li
              class="titles-list__item"
            >
              <h4>
                <routerlink-stub
                  class="titles-list__link titles-list__link--loading"
                  data-testid="ssa-XVIII-18"
                  to="[object Object]"
                >
                   Title XVIII 
                </routerlink-stub>
              </h4>
            </li>
            <li
              class="titles-list__item"
            >
              <h4>
                <routerlink-stub
                  class="titles-list__link titles-list__link--active titles-list__link--loading"
                  data-testid="ssa-XIX-19"
                  to="[object Object]"
                >
                   Title XIX 
                </routerlink-stub>
              </h4>
            </li>
            <li
              class="titles-list__item"
            >
              <h4>
                <routerlink-stub
                  class="titles-list__link titles-list__link--loading"
                  data-testid="ssa-XXI-21"
                  to="[object Object]"
                >
                   Title XXI 
                </routerlink-stub>
              </h4>
            </li>
          </ul>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="acts__list"
    >
      <li
        class="acts-list__item"
      >
        <h4
          class="acts-item__heading acts-item__heading--active"
        >
           Social Security Act 
        </h4>
        <ul
          class="titles__list"
        >
          <li
            class="titles-list__item"
          >
            <h4>
              <routerlink-stub
                class="titles-list__link titles-list__link--loading"
                data-testid="ssa-XI-11"
                to="[object Object]"
              >
                 Title XI 
              </routerlink-stub>
            </h4>
          </li>
          <li
            class="titles-list__item"
          >
            <h4>
              <routerlink-stub
                class="titles-list__link titles-list__link--loading"
                data-testid="ssa-XVI-16"
                to="[object Object]"
              >
                 Title XVI 
              </routerlink-stub>
            </h4>
          </li>
          <li
            class="titles-list__item"
          >
            <h4>
              <routerlink-stub
                class="titles-list__link titles-list__link--loading"
                data-testid="ssa-XVIII-18"
                to="[object Object]"
              >
                 Title XVIII 
              </routerlink-stub>
            </h4>
          </li>
          <li
            class="titles-list__item"
          >
            <h4>
              <routerlink-stub
                class="titles-list__link titles-list__link--active titles-list__link--loading"
                data-testid="ssa-XIX-19"
                to="[object Object]"
              >
                 Title XIX 
              </routerlink-stub>
            </h4>
          </li>
          <li
            class="titles-list__item"
          >
            <h4>
              <routerlink-stub
                class="titles-list__link titles-list__link--loading"
                data-testid="ssa-XXI-21"
                to="[object Object]"
              >
                 Title XXI 
              </routerlink-stub>
            </h4>
          </li>
        </ul>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "isUnmounted": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "unmount": [Function],
  "updateProps": [Function],
}
`;

exports[`Statute Table Selector > SSA table type > Creates a snapshot of the Statute Selector with default props 1`] = `
{
  "baseElement": <body>
    <div>
      <ul
        class="acts__list"
      >
        <li
          class="acts-list__item"
        >
          <h4
            class="acts-item__heading acts-item__heading--active"
          >
             Social Security Act 
          </h4>
          <ul
            class="titles__list"
          >
            <li
              class="titles-list__item"
            >
              <h4>
                <routerlink-stub
                  class="titles-list__link"
                  data-testid="ssa-XI-11"
                  to="[object Object]"
                >
                   Title XI 
                </routerlink-stub>
              </h4>
            </li>
            <li
              class="titles-list__item"
            >
              <h4>
                <routerlink-stub
                  class="titles-list__link"
                  data-testid="ssa-XVI-16"
                  to="[object Object]"
                >
                   Title XVI 
                </routerlink-stub>
              </h4>
            </li>
            <li
              class="titles-list__item"
            >
              <h4>
                <routerlink-stub
                  class="titles-list__link"
                  data-testid="ssa-XVIII-18"
                  to="[object Object]"
                >
                   Title XVIII 
                </routerlink-stub>
              </h4>
            </li>
            <li
              class="titles-list__item"
            >
              <h4>
                <routerlink-stub
                  class="titles-list__link titles-list__link--active"
                  data-testid="ssa-XIX-19"
                  to="[object Object]"
                >
                   Title XIX 
                </routerlink-stub>
              </h4>
            </li>
            <li
              class="titles-list__item"
            >
              <h4>
                <routerlink-stub
                  class="titles-list__link"
                  data-testid="ssa-XXI-21"
                  to="[object Object]"
                >
                   Title XXI 
                </routerlink-stub>
              </h4>
            </li>
          </ul>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="acts__list"
    >
      <li
        class="acts-list__item"
      >
        <h4
          class="acts-item__heading acts-item__heading--active"
        >
           Social Security Act 
        </h4>
        <ul
          class="titles__list"
        >
          <li
            class="titles-list__item"
          >
            <h4>
              <routerlink-stub
                class="titles-list__link"
                data-testid="ssa-XI-11"
                to="[object Object]"
              >
                 Title XI 
              </routerlink-stub>
            </h4>
          </li>
          <li
            class="titles-list__item"
          >
            <h4>
              <routerlink-stub
                class="titles-list__link"
                data-testid="ssa-XVI-16"
                to="[object Object]"
              >
                 Title XVI 
              </routerlink-stub>
            </h4>
          </li>
          <li
            class="titles-list__item"
          >
            <h4>
              <routerlink-stub
                class="titles-list__link"
                data-testid="ssa-XVIII-18"
                to="[object Object]"
              >
                 Title XVIII 
              </routerlink-stub>
            </h4>
          </li>
          <li
            class="titles-list__item"
          >
            <h4>
              <routerlink-stub
                class="titles-list__link titles-list__link--active"
                data-testid="ssa-XIX-19"
                to="[object Object]"
              >
                 Title XIX 
              </routerlink-stub>
            </h4>
          </li>
          <li
            class="titles-list__item"
          >
            <h4>
              <routerlink-stub
                class="titles-list__link"
                data-testid="ssa-XXI-21"
                to="[object Object]"
              >
                 Title XXI 
              </routerlink-stub>
            </h4>
          </li>
        </ul>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "isUnmounted": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "unmount": [Function],
  "updateProps": [Function],
}
`;
